fundamentals/streaming/DynamicBinaryStream/DynamicBinaryStream/DynamicBinaryStream.csproj

```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0-windows7</TargetFramework>
```

Comment:
Without `-windows7`, I get this when building:
```
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(18,21,18,58): warning CA1416: This call site is reachable on all platforms. 'Font' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(33,26,33,61): warning CA1416: This call site is reachable on all platforms. 'Bitmap' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(40,55,40,68): warning CA1416: This call site is reachable on all platforms. 'Brushes.Black' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(38,17,38,108): warning CA1416: This call site is reachable on all platforms. 'Graphics.FillRectangle(Brush, int, int, int, int)' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(41,66,41,78): warning CA1416: This call site is reachable on all platforms. 'Brushes.Gray' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(41,17,41,106): warning CA1416: This call site is reachable on all platforms. 'Graphics.DrawString(string?, Font, Brush, float, float)' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(35,35,35,61): warning CA1416: This call site is reachable on all platforms. 'Graphics.FromImage(Image)' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(40,17,40,82): warning CA1416: This call site is reachable on all platforms. 'Graphics.DrawString(string?, Font, Brush, float, float)' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Model\FrameSourceAsync.cs(36,42,36,69): warning CA1416: This call site is reachable on all platforms. 'SolidBrush' is only supported on: 'windows'.
1>C:\code\AspNetCore.Docs.Samples\fundamentals\streaming\DynamicBinaryStream\Results\AnimationResult.cs(25,45,25,105): warning CA1416: This call site is reachable on all platforms. 'Bitmap' is only supported on: 'windows'.
```
Should I still be indicating that it targets all platforms?

Reply:
It would be better to update the sample to not use platform specific APIs.

Reply:
Okay. Do you have a proposal for an API that does image manipulation including text rendering? I haven't done much of that. Not sure what all is available from platform-independent .NET.

Reply:
Oh, also, the AnimatedGif NuGet package has a hard dependency on System.Drawing.Common. It would be a lot of work to get rid of that dependency.

Reply:
@Rick-Anderson is there any guidance about this repo having platform specific samples? E.g. these only work on Windows.

Reply:
> @Rick-Anderson is there any guidance about this repo having platform specific samples? E.g. these only work on Windows.

Unless you have a good reason, the sample shouldn't be limited to Windows.

Reply:
> It would be better to update the sample to not use platform specific APIs.

I know it's ugly, but what about 
```
 if(System.Runtime.InteropServices.RuntimeInformation
                                               .IsOSPlatform(OSPlatform.Windows));
 // Windows code
```

Reply:
The trouble is that the sample doesn't work without it, there's no else alternative for the other platforms.

Reply:
The Mono project has `libgdiplus` and a `System.Drawing` implementation on top of it. I wonder if that's accessible on non-Windows builds??

Reply:
I have been investigating options.

* Microsoft used to support the use of `libgdiplus` on non-Windows platforms, having imported the code from the Mono project. However, after reported bugs went unfixed for some time, this support was removed.
* There is a library called SixLabors.ImageSharp. It does include code for processing indexed images. However, it is internal-only functionality used during the saving process for file formats like GIF, and not exposed externally.
* There is a library called SkiaSharp. It used to support indexed colour formats, but as of 4 years ago it does not any more.
* There is a library called .NET MAUI. However, it appears to be built on top of other libraries and exposes a least common denominator. On Linux, the only supported back-end is SkiaSharp.

Do you know of any other options?

Reply:
Hmm, I might be able to propose an adjustment to `AnimatedGif` that offloads the indexed image processing entirely so that libraries like ImageSharp and SkiaSharp might potentially be usable after all... Investigating.

Reply:
Okay I'm pretty sure SkiaSharp isn't an option on account of it not having any GIF write support at all.

Reply:
I've created a PR on `AnimatedGif` that allows for the underlying image library to be abstracted, and created an abstraction that uses `SixLabors.ImageSharp`. It seems to work in my tests. If it is accepted upstream, then the sample can be switched to that instead of `System.Drawing.Common`, which should allow it to work on Linux.

https://github.com/mrousavy/AnimatedGif/pull/19